nilearn/decomposition/tests/test_dict_learning.py

```diff
@@ -1,137 +1,78 @@
 import numpy as np
 import pytest
-from nibabel import Nifti1Image
-from sklearn.utils.estimator_checks import parametrize_with_checks
 
-from nilearn._utils.estimator_checks import (
-    check_estimator,
-    nilearn_check_estimator,
-    return_expected_failed_checks,
-)
-from nilearn._utils.tags import SKLEARN_LT_1_6
-from nilearn._utils.testing import write_imgs_to_path
-from nilearn.conftest import _affine_eye
 from nilearn.decomposition.dict_learning import DictLearning
-from nilearn.decomposition.tests.test_canica import _make_canica_test_data
+from nilearn.decomposition.tests.conftest import (
+    RANDOM_STATE,
+    check_decomposition_estimator,
+)
 from nilearn.image import get_data, iter_img
 from nilearn.maskers import NiftiMasker
-
-SHAPE = (30, 30, 5)
-
-
-@pytest.fixture(scope="module")
-def mask_img():
-    mask = np.ones(SHAPE)
-    mask[:5] = 0
-    mask[-5:] = 0
-    mask[:, :5] = 0
-    mask[:, -5:] = 0
-    mask[..., -2:] = 0
-    mask[..., :2] = 0
-    return Nifti1Image(mask, _affine_eye())
-
-
-@pytest.fixture(scope="module")
-def canica_data():
-    """Create a canonical ICA data for testing purposes."""
-    return _make_canica_test_data()[0]
-
-
-ESTIMATORS_TO_CHECK = [DictLearning()]
-
-if SKLEARN_LT_1_6:
-
-    @pytest.mark.parametrize(
-        "estimator, check, name",
-        check_estimator(estimators=ESTIMATORS_TO_CHECK),
-    )
-    def test_check_estimator_sklearn_valid(estimator, check, name):  # noqa: ARG001
-        """Check compliance with sklearn estimators."""
-        check(estimator)
-
-    @pytest.mark.xfail(reason="invalid checks should fail")
-    @pytest.mark.parametrize(
-        "estimator, check, name",
-        check_estimator(
-            estimators=ESTIMATORS_TO_CHECK,
-            valid=False,
-        ),
-    )
-    def test_check_estimator_sklearn_invalid(estimator, check, name):  # noqa: ARG001
-        """Check compliance with sklearn estimators."""
-        check(estimator)
-
-else:
-
-    @parametrize_with_checks(
-        estimators=ESTIMATORS_TO_CHECK,
-        expected_failed_checks=return_expected_failed_checks,
-    )
-    def test_check_estimator_sklearn(estimator, check):
-        """Check compliance with sklearn estimators."""
-        check(estimator)
-
-
-@pytest.mark.parametrize(
-    "estimator, check, name",
-    nilearn_check_estimator(estimators=ESTIMATORS_TO_CHECK),
-)
-def test_check_estimator_nilearn(estimator, check, name):  # noqa: ARG001
-    """Check compliance with nilearn estimators rules."""
-    check(estimator)
+from nilearn.surface.surface import get_data as get_surface_data
 
 
+@pytest.mark.parametrize("data_type", ["nifti"])
```

Comment:
we could technically run this for surface too no? Or is it too expensive as a smoke test to run twice?